docs/demos.md

```diff
@@ -0,0 +1,29 @@
+# AMT Demos 
+
+These documentations give an overview of how the demos work and explain some mechanism in these demos. (including some functions we use in the demo)
+
+For a better understanding, we suggest that you can try to revise some code to change and verify the behaviour of the demos or dig into them by using commands like `make debug` and trace them one by one.
+
+* [yielding](../demos/yielding/README.md)
+* [exyielding](../demos/exyielding/README.md)
+* [message](../demos/message/README.md)
+* [exit](../demos/exit/README.md)
+* [spawn](../demos/spawn/README.md)
+* [stackcheck](../demos/stackcheck/README.md)
+* [args](../demos/args/README.md)
+* [mutexes](../demos/mutexes/README.md)
+* [timer](../demos/timer/README.md)
+* [threadlocalstorage](../demos/threadlocalstorage/README.md)
+* [conditionvariables](../demos/conditionvariables/README.md)
+* [cancel](../demos/cancel/README.md)
+* [file](../demos/file/README.md)
+* [alloc](../demos/alloc/README.md)
+* [loadbinary](../demos/loadbinary/README.md)
+* [loadbinaries](../demos/loadbinaries/README.md)
+* [loadpiebinary](../demos/loadpiebinary/README.md)
+* [parentchild](../demos/parentchild/README.md)
+* [permissions](../demos/permissions/README.md)
+* [trace](../demos/trace/README.md)
+* [signalhandling](../demos/signalhandling/README.md)
+* [backtrace](../demos/backtrace/README.md)
+* [fibres](../demos/fibres/README.md)
```

Comment:
I'd rather you list just the ones that have been done. (assume that I'll merge this after spawn goes in)

Reply:
(but if you move the table here you can just link the first few names)